src/test/java/org/broadinstitute/hellbender/tools/dataflow/SmallBamWriterTest.java

```diff
@@ -0,0 +1,172 @@
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SamInputResource;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.ValidationStringency;
+import org.broadinstitute.hellbender.dev.pipelines.bqsr.SmallBamWriter;
+import org.broadinstitute.hellbender.engine.dataflow.ReadsSource;
+import org.broadinstitute.hellbender.tools.IntegrationTestSpec;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.dataflow.BucketUtils;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.Channels;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tests for SmallBamWriter.
+ */
+public class SmallBamWriterTest extends BaseTest {
+    private static final String LOCAL_INPUT = "src/test/resources/org/broadinstitute/hellbender/engine/reads_data_source_test2.bam";
+    private static final String THIS_TEST_FOLDER = "org/broadinstitute/hellbender/engine/reads_data_source_test2.bam";
+
+    private SAMFileHeader header;
+
+    @Test
+    public void checkLocal() throws Exception {
+        File out = createTempFile("temp",".bam");
+        String outputFile = out.getPath();
+        testReadAndWrite(LOCAL_INPUT, outputFile, false);
+    }
+
+    @Test(groups = {"bucket"})
+    public void checkGCSInput() throws Exception {
+        File out = createTempFile("temp",".bam");
+        String outputFile = out.getPath();
+        testReadAndWrite(getCloudInput(), outputFile, false);
+    }
+
+    // this leaves output files around, for now.
+    @Test(groups = {"bucket"})
+    public void checkGCSOutput() throws Exception {
+        File out = createTempFile("temp",".bam");
+        String tempName = out.getName();
+        String outputPath = getDataflowTestStaging() + tempName;
+        testReadAndWrite(LOCAL_INPUT, outputPath, false);
+    }
+
+    // this leaves output files around, for now.
+    @Test(groups = {"cloud"})
+    public void checkCloud() throws Exception {
+        File out = createTempFile("temp",".bam");
+        String tempName = out.getName();
+        String outputPath = getDataflowTestStaging() + tempName;
+        testReadAndWrite(getCloudInput(), outputPath, true);
+    }
+
+    protected void testReadAndWrite(final String inputPath, final String outputPath, boolean remote) throws IOException {
+        String localInput = inputPath;
+        String localOutput = outputPath;
+        logger.info("outputPath="+outputPath);
+        final Pipeline pipeline = setupPipeline(remote);
+        if (BucketUtils.isCloudStorageUrl(inputPath)) {
+            File out = createTempFile("temp-input",".bam");
+            localInput = out.getPath();
+            localInput="temp-input.bam";
```

Comment:
Looks like you forgot to fix this clobbering of `localInput` (and `localOutput` as well below)